feat(html): add default lang to site metadata, pass to helmet

jendowns opened the pull request:
Closes #557

Adds default language via `lang` attribute to `HTML` element, per [Checkpoint 3.1.1.](https://www.ibm.com/able/guidelines/ci162/language_of_page.html)

#### Changelog

**New**

- add `lang` to `siteMetadata`

jendowns commented on packages/gatsby-theme-carbon/src/components/Meta.js:
```diff
@@ -34,7 +34,9 @@ const Meta = ({ pageTitle, pageDescription, pageKeywords, titleType }) => {
       ]}
-    />
+    >
+      <html lang={lang || 'en'} />
```
I think this is nice because even if someone doesn't add `lang` to their own metadata, at least their site will have a valid `lang` value 🤔 But let me know what you think!

vpicone replied:
If they don't add one, it should grab the `en` you have defined in the theme's gatsby-config along with the other defaults (title, keyword etc)

jendowns commented:
Thanks @vpicone! Just made the change you suggested above 👍

vpicone commented on packages/example/gatsby-config.js:
```diff
@@ -3,6 +3,7 @@ module.exports = {
     title: 'Gatsby Theme Carbon',
     description: 'A Gatsby theme for the carbon design system',
     keywords: 'gatsby,theme,carbon',
+    lang: 'en',
```
I think maybe we should leave this out of the example, it doubles as the starter-kit and folks might not know if they need to leave it in.

Would could update [https://gatsby-theme-carbon.now.sh/guides/configuration#site-title-and-description](this part of the config guide) to just be Site Metadata and add a note there maybe? "**If necessary**, you can change the default language of your document using the `lang` property" (maybe dont need the emphasis haha)

Some thing like 
```suggestion
```

vpicone commented:
Awesome, thanks @jendowns just had one more quick thing

jendowns commented:
Thanks @vpicone! I just updated the example/demo site's metadata & also added some info to the configuration guide. Could you take another look when you get a chance?
